Dev

paulthomasjerome opened the pull request:
This encompasses everything up to make pull request on the Shift Bid Trello board

paulthomasjerome committed:
Finish adding react

paulthomasjerome committed:
Push for pull request to blm

paulthomasjerome committed:
RM X Files

beaulm commented on .gitignore:
```diff
@@ -1 +1,7 @@
+
+.yo-rc.json
+
+.babelrc
```
If someone else pulls down the project and tries to build it, how will babel and webpack know what to do without their configuration files?

beaulm commented on dist/index.html:
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
```
This file is just an artifact from the build process and not actually used, correct?

If so, maybe we can either:

1. Fix the build process to not create this file
1. Use .gitignore to exclude it from the repo

beaulm commented on package.json:
```diff
@@ -3,14 +3,39 @@
     "test": "echo \"Error: no test specified\" && exit 1",
-    "start": "nodemon index.js"
+    "start": "nodemon service.js",
+    "build": "webpack --mode development"
```
Some day we'll want another script like:
`"build": "webpack --mode production"` or something like that

beaulm commented on package.json:
```diff
@@ -3,14 +3,39 @@
+    "webpack": "^4.39.1",
+    "webpack-cli": "^3.3.6",
+    "webpack-dev-server": "^3.7.2",
+    "wepack-cli": "0.0.1-security"
```
Hmm, did you install this package or did another package install it? It's got a very suspicious name...

paulthomasjerome replied:
Another pacakge installed it

beaulm replied:
Really, which one?

I'm highly suspicious of this package and you should be too. It's worth figuring out how it got in your dependencies list and whether or not you really want to have it there.

paulthomasjerome replied:
https://www.npmjs.com/package/wepack-cli?activeTab=readme

This was malicious and I have uninstalled it (sorry it took me so long to figure out, I am an idiot and did not realize it said _**we**_pack, I have uninstalled it but I am anxious about it so lets take a look at it when we hang out

beaulm commented on src/App.js:
```diff
@@ -0,0 +1,20 @@
+
+import React from 'react';
+
+let Welcome = props => {
```
I *think* you need to use regular function syntax for React components. For example something like:
```javascript
import React from 'react';

function Welcome({name}) {
    return <h1>Hey {name} Automating Shifts For Your Organization Like No Other</h1>
}

function App() {
    return (
        <div className="App">
            <Welcome name="Adam" />
            <Welcome name="Steve" />
            <Welcome name="Jib" />
        </div>
    )
}

export default App;
```

I haven't actually tried that code yet so I'm not sure if it's correct, but something like that at least.

paulthomasjerome replied:
This way worked but I have rewrote it with the normal function syntax

beaulm commented on src/App.js:
```diff
@@ -0,0 +1,20 @@
+
```
Remove these 2 extra lines

beaulm commented on src/App.js:
```diff
@@ -0,0 +1,20 @@
+
+const App = () => {
+    return (
+      <div className="App">
```
You have mixed indentation here

beaulm commented on src/App.js:
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+
+let Welcome = props => {
+    return <h1>Hey {props.name} Automating Shifts For Your Organization Like No Other</h1>
```
I think you need `return <h1>{`Hey ${name} Automating Shifts For Your Organization Like No Other`}</h1>;`

beaulm reviewed (commented):
Lookin good man! Great first PR. Mostly some minor things to fix up. Also, we should add linting to the project. I'll add it to the Trello board

beaulm commented on src/App.js:
```diff
@@ -0,0 +1,20 @@
+      </div>
+    );
+}
+
```
[Add prop-type checking](https://www.npmjs.com/package/prop-types)

paulthomasjerome committed:
Update App.js and remove configs from gitignore

paulthomasjerome committed:
Add ignored configs

paulthomasjerome committed:
Fix minore formatting issues

paulthomasjerome committed:
Refactor components to be functional

paulthomasjerome committed:
Add prop-types

beaulm commented on .gitignore:
```diff
@@ -1 +1,5 @@
+
+/dist/index.html
+
+/src/App.css
```
Is this file produced by the build process or used by the build process? Since it's in `/src/` I'm guessing it's used by the build process and necessary for the project.

As a corollary, I'm guessing everything in `/dist/` is just a build artifact and doesn't need to be shipped with the project since it will be created locally when someone runs `npm run build`. As such you could probably exclude `/dist/*`

paulthomasjerome replied:
That makes sense, I will add it to `.gitignore`

paulthomasjerome replied:
by it i mean `/dist/*`

beaulm commented on .yo-rc.json:
```diff
@@ -0,0 +1,76 @@
+{
```
https://stackoverflow.com/questions/49276878/can-i-delete-yo-rc-json-after-project-has-been-generated

paulthomasjerome replied:
Done

beaulm commented on src/App.js:
```diff
@@ -0,0 +1,61 @@
+`;
+
+const LoginWrapper = styled.div`
+align-items: center;
```
Your linter would have caught this indentation problem so I didn't have to

paulthomasjerome replied:
Sorry :( when i push tonight it will have linting

beaulm commented on src/App.js:
```diff
@@ -0,0 +1,61 @@
+function App() {
+  return (
+    <div className="App">
+    <MainLogin className="App"></MainLogin>
```
Is this component nested inside the div? It's hard to tell from the indentation.

Also, I can't see your CSS since you didn't check it into the project, but I'm curious what could possibly be in the `.App` class that you'd want on both the wrapper and the login form. Also, it's pretty standard to use camelCase for css class names

paulthomasjerome replied:
this was a mistake which should not have been checked in

beaulm reviewed (commented):
Still haven't had a chance to pull it down and build it yet, but you're making good progress

paulthomasjerome committed:
finally organized compnents to the point where we do not fail to build

paulthomasjerome committed:
Create ShiftItem component, Style body to override UAS, Update MainLogin component

beaulm commented on package.json:
```diff
@@ -3,14 +3,42 @@
+    "html-webpack-plugin": "^3.2.0",
+    "postcss-loader": "^3.0.0",
+    "precss": "^4.0.0",
+    "prop-types": "^15.7.2",
```
You probably don't need this in both the `dependencies` and `devDependencies`. You should remove this one

paulthomasjerome replied:
removed from `dependencies`

paulthomasjerome replied:
should have said `devDependencies`

beaulm commented on package.json:
```diff
@@ -3,14 +3,42 @@
+    "babel-plugin-syntax-dynamic-import": "^6.18.0",
+    "css-loader": "^3.2.0",
+    "html-webpack-plugin": "^3.2.0",
+    "postcss-loader": "^3.0.0",
```
Is this actually used?

paulthomasjerome replied:
Not right now, I can remove it

beaulm commented on package.json:
```diff
@@ -3,14 +3,42 @@
+    "style-loader": "^1.0.0",
+    "webpack": "^4.39.1",
+    "webpack-cli": "^3.3.6",
+    "webpack-dev-server": "^3.7.2",
```
I don't believe you need this either. We're not going to be using `webpack-dev-server`

paulthomasjerome replied:
Removed

beaulm commented on public/main.css:
```diff
@@ -0,0 +1,6 @@
+body {
+  margin: 0 auto;
```
What's the purpose of this?

paulthomasjerome replied:
Sorry this should not have been checked in, i was just experimenting with some things, the reason for this was to center the body when i changed the viewport width

beaulm commented on public/main.css:
```diff
@@ -0,0 +1,6 @@
+body {
+  margin: 0 auto;
+  background: purple;
+  width: 95vw;
```
Or this or the `height`?

paulthomasjerome replied:
Sorry this should not have been checked in, i was just experimenting with some things

beaulm commented on src/App.js:
```diff
@@ -0,0 +1,14 @@
+function App() {
+  return (
+    <div className="App">
+    <MainLogin />
```
Indent me

paulthomasjerome replied:
indented

beaulm commented on src/App.js:
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import MainLogin from './Components/MainLogin.js';
+import ShiftItem from './Components/ShiftItem.js';
```
You're importing this but never using it

paulthomasjerome replied:
Sorry this was something that should not have been checked in

beaulm commented on src/Components/MainLogin.js:
```diff
@@ -0,0 +1,46 @@
+      <LoginForm>
+        <input type="text"></input>
+        <input type="text"></input>
+        <LoginButton>
```
You should either make this line `<LoginButton>Login</LoginButton>` to be consistent with `<LoginWelcome>Login</LoginWelcome>` above, or split that out into 3 lines like this

paulthomasjerome replied:
Split <LoginWelcome>Login</LoginWelcome> into three lines

beaulm commented on src/Components/ShiftItem.js:
```diff
@@ -0,0 +1,30 @@
+`;
+
+const ShiftWrapper = styled.div`
+  background-color: cornsilk;
```
'Learn something new every day

paulthomasjerome replied:
cornsilk? haha

beaulm reviewed (commented):
Alright, I finally had a chance to pull this down and run it locally. Great start man!

I'm impressed with how quickly you picked up `styled-components`, and I like the general way you've got your project structured.

I'm not going to really critique the UI since it's obviously a work in progress and I don't think any major refactors are necessary at this time. I'd suggest the following course of action:

1. Setup linting (see Trello)
1. Make sure this project passes linting
1. Address all my comments
1. Scrub the code of anything unnecessary

I'd do all that before writing any new code. If we don't get this merged in soon it will become unwieldy. Likewise with removing unnecessary code; don't live with broken windows. It's totally legit to have unused code while developing locally, but don't check that in. Keeping your code to the minimal amount necessary will make it _much_ easier to read, reason about, and maintain, and it will also improve the performance for your users.

Your project seems to build and run just fine with only the following devDependencies
```
    "@babel/core": "^7.5.5",
    "@babel/preset-env": "^7.5.5",
    "@babel/preset-react": "^7.0.0",
    "babel-loader": "^8.0.6",
    "babel-plugin-styled-components": "^1.10.6",
    "react": "^16.9.0",
    "react-dom": "^16.9.0",
    "webpack": "^4.39.1",
    "webpack-cli": "^3.3.6"
```

Webpack explicitly requests `autoprefixer`, `precss`, and `html-webpack-plugin`, so those ones may be worth keeping, or you could just re-add them if/when you actually use them. If you _think_ you might need a package for your project and just wanna try it out, you can do an `npm install packagename` without any `--save` or `--save-dev` and then do an install with a `--save/-dev` if you want to keep it

paulthomasjerome committed:
address all of beau's comments, add linting, configure linting (basic)

paulthomasjerome commented:
I have addressed everything and added linting but i need some help with configuring linting due to issues parsing jsx, I have not added unit tests or code coverage at this time, that will be my next branch

beaulm reviewed (approved):
Looks great Paul! If you commit the .gitignore file first, it will exclude the things in /dist/ for you. You can also manually remove public/main.css if you're not using it.

I wouldn't worry too much about all that though. Worst comes to work you can just remove the files that don't need to be in the repo from a future commit.

Great work man!!
